.github/workflows/smoke.yml

```diff
@@ -283,11 +313,29 @@ jobs:
       shell: bash
       env:
         UPSTREAM_JUJU_TAG: ${{ steps.vars.outputs.upstream-juju-version }}
+        CURRENT_STABLE_JUJU_TAG: ${{ steps.vars.outputs.current-stable-juju-version }}
         UPGRADE_PARAMS_localhost: ""
-        UPGRADE_PARAMS_microk8s: ""
+        UPGRADE_PARAMS_microk8s: "--agent-stream=develop --debug"
       run: |
         set -euxo pipefail
 
+        # upgrade to the latest stable.
+        juju upgrade-controller --debug
+        
+        attempt=0
+        while true; do
+          UPDATED=$((juju show-controller --format=json || echo "") | jq -r '.c.details."agent-version"')
+          if [[ $UPDATED == $CURRENT_STABLE_JUJU_TAG* ]]; then
+              break
+          fi
+          sleep 10
+          attempt=$((attempt+1))
+          if [ "$attempt" -eq 48 ]; then
+              echo "Upgrade controller timed out"
+              exit 1
+          fi
+        done
+        
         juju upgrade-controller ${UPGRADE_PARAMS_${{ matrix.model_type }}}
```

Comment:
This part should be removed now right?

Reply:
Nah, we have to upgrade to stable first then local version.